packages/cadl-ranch-specs/http/projection/client-encoded-name/main.tsp

```diff
@@ -0,0 +1,178 @@
+    defaultName: boolean;
+  }
+
+  model JsonAndClientClientNameAndEncodedNameModel {
+    @doc("Pass in true")
+    @clientName("clientName")
+    @encodedName("application/json", "wireName")
+    defaultName: boolean;
+  }
+
+  @scenario
+  @scenarioDoc("""
+  Testing that we can project the JSON name on the wire from defaultName -> wireName.
+  Your generated SDK should generate JsonClientNameAndEncodedNameModel with one property `defaultName` with wire name `wireName`.
+
+  Expected request body:
+  ```json
+  {"wireName": true}
+  ```
+  """)
+  @route("/json")
+  @post
+  op json(...JsonClientNameAndEncodedNameModel): NoContentResponse;
+
+  @scenario
+  @scenarioDoc("""
+Testing that we can project the client name in our generated SDKs.
+Your generated SDK should generate ClientClientNameAndEncodedNameModel with one property `clientName` with wire name `defaultName`.
+
+Expected request body:
+```json
+{"defaultName": true}
+```
+""")
+  @route("/client")
+  @post
+  op client(...ClientClientNameAndEncodedNameModel): NoContentResponse;
+
+  @scenario
+  @scenarioDoc("""
+Testing that we can project the language specific name in our generated SDKs.
+Your generated SDK should generate ClientClientNameAndEncodedNameModel with one property with your language specific property name and wire name `defaultName`.
+
+Expected request body:
+```json
+{"defaultName": true}
+```
+""")
+  @route("/language")
+  @post
+  op language(...LanguageClientNameAndEncodedNameModel): NoContentResponse;
+
+  @scenario
+  @scenarioDoc("""
+Testing that we can project the client name and the wire name.
+Your generated SDK should generate JsonAndClientClientNameAndEncodedNameModel with one property with client name `clientName` and wire name `wireName`.
+
+Expected request body:
+```json
+{"wireName": true}
+```
+""")
+  @route("/json-and-client")
+  @post
+  op jsonAndClient(...JsonAndClientClientNameAndEncodedNameModel): NoContentResponse;
+}
+
+// May add query/header project after https://github.com/microsoft/cadl/issues/1487
+
+@scenario
+@scenarioDoc("""
+Testing that we can project the operation name.
+Your generated SDK should generate an operation called `clientName`.
+
+Expected status code: 204
+""")
+@route("/operation")
+@clientName("clientName")
+@post
+op operation(): NoContentResponse;
+
+@scenario
+@scenarioDoc("""
+Testing that we can project a parameter name.
+Your generated SDK should generate an operation `parameter` with a single parameter called `clientName`.
+
+Expected query parameter: `default-name="true"`
+
+""")
+@route("/parameter")
+@post
+op parameter(
+  @clientName("clientName")
+  @query
+  `default-name`: string,
+): NoContentResponse;
+
+@route("/model")
+@operationGroup
+namespace Model {
+  @clientName("CSModel", "csharp")
+  @clientName("GoModel", "go")
+  @clientName("JavaModel", "java")
+  @clientName("TSModel", "javascript")
+  @clientName("PythonModel", "python")
+  model ModelWithLanguageClientNameAndEncodedName {
```

Comment:
Not sure this name make sense has it has no `@encodedName`

Reply:
does model name make sense in wire layer ? I feel like no?

Reply:
It doesn’t yeah I am just saying the model names needs to be changed

Reply:
Got your point, updated.